uwblueprint/rowan-house
# [RHS-130]: Add email verification workflow and authentication

## Implementation Description
After signing up, users that have not verified should not be able to access any private pages until they have verified their email. As such, we have created a page after sign up that reminds users to verify their email and all PrivateRoutes redirect to this page until they have verified. Furthermore, we have appended an email verification status property to the now termed AuthUser, adding backend support for this property.

## Screenshots
<img width="1505" alt="image" src="https://user-images.githubusercontent.com/53355975/171554461-410e74d5-cd0d-436b-89e8-83c4a9329b25.png">

## What should reviewers focus on?
- See if it works correctly
- Is the user flow intuitive?

## Testing Procedure
- Go through signup flow and verify your email, then reload page to see if it works
- Try going to different pages after going through the signup flow to see what you can reach while "logged in" but not verified

## Things to Note
What should be the user flow for people who signup but want to leave the page, then try to return to login later? This is avoided when you let them in without having them verify. One thought I have is logging out as soon as you leave the verify email page without verifying your email, but a discussion may need to be had about this.

## Checklist
